app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers.php

```diff
@@ -578,6 +428,288 @@ protected function addError($errorMessage)
+            case 'shipping_company':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer);
+
+                if ($address) {
+                    $company = $address->getCompany();
+                    if ($company) {
+                        $mergeVars[$key] = $company;
+                    }
+                }
+                break;
+            case 'billing_telephone':
+            case 'shipping_telephone':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer);
+
+                if ($address) {
+                    $telephone = $address->getTelephone();
+                    if ($telephone) {
+                        $mergeVars[$key] = $telephone;
+                    }
+                }
+                break;
+            case 'billing_country':
+            case 'shipping_country':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer);
+
+                if ($address) {
+                    $countryCode = $address->getCountry();
+                    if ($countryCode) {
+                        $countryName = Mage::getModel('directory/country')->loadByCode($countryCode)->getName();
+                        $mergeVars[$key] = $countryName;
+                    }
+                }
+                break;
+            case 'billing_zipcode':
+            case 'shipping_zipcode':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer);
+
+                if ($address) {
+                    $zipCode = $address->getPostcode();
+                    if ($zipCode) {
+                        $mergeVars[$key] = $zipCode;
+                    }
+                }
+                break;
+            case 'billing_state':
+            case 'shipping_state':
+                $address = $this->getAddressForCustomizedAttributes($customAtt, $customer);
+
+                if ($address) {
+                    $state = $address->getRegion();
+                    if ($state) {
+                        $mergeVars[$key] = $state;
+                    }
+                }
+                break;
+            case 'dop':
+                $dop = $this->getLastDateOfPurchase($subscriberEmail);
+                if ($dop) {
+                    $mergeVars[$key] = $dop;
+                }
+                break;
+            case 'store_code':
+                $storeCode = Mage::getModel('core/store')->load($storeId)->getCode();
+                $mergeVars[$key] = $storeCode;
+        }
+
+        return $mergeVars;
+    }
+
+    /**
+     * @param $subscriber
+     * @param $attributeCode
+     * @param $customer
+     * @param $mergeVars
+     * @param $key
+     * @param $storeId
+     * @param $attribute
+     * @return array | returns an array of mergeFields.
+     */
+    protected function customerAttributes($subscriber, $attributeCode, $customer, $mergeVars, $key, $storeId, $attribute)
```

Comment:
524 | WARNING | [ ] Function's cyclomatic complexity (20) exceeds
     |         |     10; consider refactoring the function
 524 | WARNING | [ ] Line exceeds 120 characters; contains 121
     |         |     characters
 534 | WARNING | [ ] count(...) function should not be used to check
     |         |     if array is empty. Use empty(...) language
     |         |     construct instead
 546 | WARNING | [ ] Variable "group_id" is not in valid camel caps
     |         |     format
 548 | WARNING | [ ] Variable "group_id" is not in valid camel caps
     |         |     format
 587 | WARNING | [x] No blank line found after control structure
 588 | WARNING | [ ] Line exceeds 120 characters; contains 122
     |         |     characters
 588 | WARNING | [ ] This comment is 70% valid code; is this
     |         |     commented out code?